Cut the size of JSGlobalObject.h down

darinadler opened the pull request:
#### 568657736a638d4a2649e98fa9454ce3cc042dd1
<pre>
Cut the size of JSGlobalObject.h down
<a href="https://bugs.webkit.org/show_bug.cgi?id=253376">https://bugs.webkit.org/show_bug.cgi?id=253376</a>
rdar://problem/106225153

Reviewed by Jer Noble.

Moved things out of JSGlobalObject.h, and removed unneeded includes of JSGlobalObject.h.

* Source/JavaScriptCore/API/APICast.h: Updated includes.

* Source/JavaScriptCore/API/JSAPIGlobalObject.cpp:
(JSC::JSAPIGlobalObject::globalObjectMethodTable): Use function for this class&apos;s
globalObjectMethodTable so header doesn&apos;t need to include GlobalObjectMethodTable.h.
* Source/JavaScriptCore/API/JSAPIGlobalObject.h: Ditto.
* Source/JavaScriptCore/API/JSAPIGlobalObject.mm:
(JSC::JSAPIGlobalObject::globalObjectMethodTable): Ditto.

* Source/JavaScriptCore/API/JSCallbackObjectFunctions.h: Updated includes.

* Source/JavaScriptCore/API/JSClassRef.cpp:
(OpaqueJSClass::contextData): Use new JSGlobalObject::contextData function, helping hide
more of the details of the JSGlobalObject rare data from other includers of JSGlobalObject.h

* Source/JavaScriptCore/CMakeLists.txt: Added GlobalObjectMethodTable.h.
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj: Ditto.

* Source/JavaScriptCore/bytecode/BytecodeIntrinsicRegistry.cpp: Updated includes.
* Source/JavaScriptCore/bytecode/CodeBlock.h: Ditto.
* Source/JavaScriptCore/bytecode/ExecutionCounter.h: Ditto.
* Source/JavaScriptCore/bytecode/ObjectAllocationProfile.h: Ditto.
* Source/JavaScriptCore/bytecode/SpeculatedType.cpp: Ditto.
* Source/JavaScriptCore/bytecompiler/NodesCodegen.cpp: Ditto.
* Source/JavaScriptCore/debugger/DebuggerEvalEnabler.h: Ditto.
* Source/JavaScriptCore/debugger/ScriptProfilingScope.h: Ditto.
* Source/JavaScriptCore/dfg/DFGOperations.cpp: Ditto.
* Source/JavaScriptCore/dfg/DFGStrengthReductionPhase.cpp: Ditto.
* Source/JavaScriptCore/ftl/FTLAbstractHeapRepository.cpp: Ditto.
* Source/JavaScriptCore/heap/Heap.cpp: Ditto.

* Source/JavaScriptCore/heap/IsoSubspace.h: Removed unused isAllocatedFromIsoSubspace.

* Source/JavaScriptCore/interpreter/CallFrameInlines.h: Updated includes.
* Source/JavaScriptCore/interpreter/Interpreter.cpp: Ditto.
* Source/JavaScriptCore/jsc.cpp: Ditto.
* Source/JavaScriptCore/runtime/AggregateError.h: Ditto.
* Source/JavaScriptCore/runtime/AggregateErrorConstructor.cpp: Ditto.
* Source/JavaScriptCore/runtime/AggregateErrorConstructor.h: Ditto.
* Source/JavaScriptCore/runtime/AggregateErrorPrototype.h: Ditto.
* Source/JavaScriptCore/runtime/ArrayBufferSharingMode.h: Ditto.
* Source/JavaScriptCore/runtime/ArrayPrototypeInlines.h: Ditto.
* Source/JavaScriptCore/runtime/BytecodeCacheError.h: Ditto.
* Source/JavaScriptCore/runtime/DeferredWorkTimer.cpp: Ditto.
* Source/JavaScriptCore/runtime/DirectEvalExecutable.cpp: Ditto.
* Source/JavaScriptCore/runtime/ExecutableBase.h: Ditto.
* Source/JavaScriptCore/runtime/FunctionConstructor.cpp: Ditto.

* Source/JavaScriptCore/runtime/GlobalObjectMethodTable.h: Added. Code here was moved
from JSGlobalObject.h.

* Source/JavaScriptCore/runtime/IndirectEvalExecutable.cpp: Updated includes.
* Source/JavaScriptCore/runtime/IntlObject.cpp: Ditto.

* Source/JavaScriptCore/runtime/JSArrayBufferView.h: Marked some functions inline so we
get warnings if we forget to include the JSArrayBufferViewInlines.h header, even in
debug builds.

* Source/JavaScriptCore/runtime/JSBigInt.h: Updated includes.
* Source/JavaScriptCore/runtime/JSFinalizationRegistry.cpp: Ditto.

* Source/JavaScriptCore/runtime/JSGenericTypedArrayView.h: Moved inline function bodies
out of here into JSGenericTypedArrayViewInlines.h.
* Source/JavaScriptCore/runtime/JSGenericTypedArrayViewInlines.h: Ditto.

* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::baseGlobalObjectMethodTable): Use function for this so header
doesn&apos;t need to include GlobalObjectMethodTable.h.
(JSC::JSGlobalObject::JSGlobalObject): Use baseGlobalObjectMethodTable.
(JSC::JSGlobalObject::typedArrayConstructorSpeciesAbsenceWatchpoint): Moved from header.
It is only used in this file.
(JSC::JSGlobalObject::typedArrayPrototypeSymbolIteratorAbsenceWatchpoint): Ditto.
(JSC::JSGlobalObject::typedArrayPrototypeConstructorWatchpoint): Ditto.

* Source/JavaScriptCore/runtime/JSGlobalObject.h: Updated includes and forward declarations.
Moved GlobalObjectData out of the header. Renamed JSGlobalObject::JSGlobalObjectRareData
to JSGlobalObject::RareData and moved it to the inlines header. Moved
JSGlobalObject::GlobalPropertyInfo out of the header.

* Source/JavaScriptCore/runtime/JSGlobalObjectFunctions.cpp: Updated includes.

* Source/JavaScriptCore/runtime/JSGlobalObjectInlines.h: Moved things here from JSGlobalObject.h.

* Source/JavaScriptCore/runtime/JSModuleLoader.cpp: Updated includes.
* Source/JavaScriptCore/runtime/JSONObject.cpp: Ditto.
* Source/JavaScriptCore/runtime/JSObjectInlines.h: Ditto.
* Source/JavaScriptCore/runtime/JSPromise.cpp: Ditto.
* Source/JavaScriptCore/runtime/JSScriptFetchParameters.h: Ditto.
* Source/JavaScriptCore/runtime/JSScriptFetcher.h: Ditto.
* Source/JavaScriptCore/runtime/JSSourceCode.h: Ditto.
* Source/JavaScriptCore/runtime/JSWithScope.h: Ditto.
* Source/JavaScriptCore/runtime/LazyClassStructure.h: Ditto.
* Source/JavaScriptCore/runtime/LazyProperty.h: Ditto.
* Source/JavaScriptCore/runtime/ModuleProgramExecutable.h: Ditto.
* Source/JavaScriptCore/runtime/PrototypeKey.h: Ditto.
* Source/JavaScriptCore/runtime/ProxyObject.h: Ditto.

* Source/JavaScriptCore/runtime/RegExpCachedResult.h: Moved RegExpCachedResult::record
to RegExpGlobalDataInlines.h.
* Source/JavaScriptCore/runtime/RegExpGlobalDataInlines.h: Ditto.

* Source/JavaScriptCore/runtime/ShadowRealmObject.cpp: Updated includes.
* Source/JavaScriptCore/runtime/StructureCache.h: Ditto.
* Source/JavaScriptCore/runtime/StructureInlines.h: Ditto.
* Source/JavaScriptCore/tools/JSDollarVM.cpp: Ditto.
* Source/JavaScriptCore/wasm/js/JSWebAssembly.cpp: Ditto.
* Source/JavaScriptCore/wasm/js/JSWebAssemblyHelpers.h: Ditto.

* Source/WebCore/Headers.cmake: Removed WebGLAnyInlines.h.

* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMSessionClearKey.cpp:
Updated includes.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Removed WebGLAnyInlines.h.

* Source/WebCore/bindings/js/JSCallbackData.h: Updated includes.
* Source/WebCore/bindings/js/JSDOMConvertBufferSource.h: Ditto.
* Source/WebCore/bindings/js/JSDOMConvertDate.h: Ditto.
* Source/WebCore/bindings/js/JSDOMConvertUnion.h: Ditto.
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp: Ditto.
* Source/WebCore/bindings/js/JSDOMGlobalObject.h: Ditto.
* Source/WebCore/bindings/js/JSDOMGuardedObject.h: Ditto.

* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
(WebCore::JSDOMWindowBase::globalObjectMethodTable): Use function for this class&apos;s
globalObjectMethodTable so header doesn&apos;t need to include GlobalObjectMethodTable.h.
* Source/WebCore/bindings/js/JSDOMWindowBase.h: Ditto.

* Source/WebCore/bindings/js/JSDOMWrapperCache.h: Updated includes.

* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.cpp:
(WebCore::JSRemoteDOMWindowBase::globalObjectMethodTable): Use function for this class&apos;s
globalObjectMethodTable so header doesn&apos;t need to include GlobalObjectMethodTable.h.
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.h: Ditto.
* Source/WebCore/bindings/js/JSShadowRealmGlobalScopeBase.cpp:
(WebCore::JSShadowRealmGlobalScopeBase::globalObjectMethodTable): Ditto.
* Source/WebCore/bindings/js/JSShadowRealmGlobalScopeBase.h: Ditto.
* Source/WebCore/bindings/js/JSWorkerGlobalScopeBase.cpp:
(WebCore::JSWorkerGlobalScopeBase::globalObjectMethodTable): Ditto.
* Source/WebCore/bindings/js/JSWorkerGlobalScopeBase.h: Ditto.
* Source/WebCore/bindings/js/JSWorkletGlobalScopeBase.cpp:
(WebCore::JSWorkletGlobalScopeBase::globalObjectMethodTable): Ditto.
* Source/WebCore/bindings/js/JSWorkletGlobalScopeBase.h: Ditto.

* Source/WebCore/bindings/js/ScriptWrappable.h: Add inline keyword for declaration of
some functions that are inlines in the ScriptWrappableInlines.h header.

* Source/WebCore/bindings/js/ScriptWrappableInlines.h: Updated includes.
* Source/WebCore/bindings/js/SerializedScriptValue.cpp: Ditto.
* Source/WebCore/bindings/js/WebCoreTypedArrayController.h: Ditto.
* Source/WebCore/bridge/runtime_method.h: Ditto.
* Source/WebCore/html/FormAssociatedCustomElement.h: Ditto.
* Source/WebCore/html/canvas/EXTDisjointTimerQuery.cpp: Ditto.
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp: Ditto.

* Source/WebCore/html/canvas/WebGLAnyInlines.h: Removed.

* Source/WebCore/html/canvas/WebGLRenderingContext.cpp: Updated includes.
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp: Ditto.
* Source/WebCore/html/canvas/WebGLRenderingContextBase.h: Ditto.
* Source/WebCore/html/track/DataCue.h: Ditto.
* Source/WebCore/inspector/InspectorCanvas.h: Ditto.
* Source/WebCore/page/WindowOrWorkerGlobalScope.cpp: Ditto.
* Source/WebCore/platform/SerializedPlatformDataCue.h: Ditto.
* Source/WebKit/Platform/IPC/JSIPCBinding.h: Ditto.

Canonical link: <a href="https://commits.webkit.org/261249@main">https://commits.webkit.org/261249@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ee0f3f161c2feb5bb9157f3d77de1ba06a7e16fb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111053 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20193 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43617 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/2481 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/119898 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/115011 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21568 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11295 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/86/builds/2121 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/116807 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/16012 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99196 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/103549 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/97969 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/30860 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44466 "Passed tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/99634 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12710 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/32195 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/86374 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/10817 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13253 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9178 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/100777 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18670 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/51796 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/31427 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7801 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15202 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/108814 "Built successfully") | 
| | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/26829 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4953b1300fc287f9c8c8e7a0d80efcec16ccb1d6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/11083

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/9aea5e9aedf9a5ccbcc16e5685922a5f7f120607. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/11083

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/3b413e8105033ab097921648941e2dbb8a3b2a35. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/11083

jernoble commented on Source/JavaScriptCore/API/JSAPIGlobalObject.h:
```diff
@@ -70,8 +69,9 @@ class JSAPIGlobalObject final : public JSGlobalObject {
     JSAPIGlobalObject(VM& vm, Structure* structure)
-        : Base(vm, structure, &s_globalObjectMethodTable)
+        : Base(vm, structure, globalObjectMethodTable())
     { }
```
If this constructor is private:, can it be moved into the .cpp file?

darinadler replied:
It can, yes. In fact I moved it in an earlier version of the patch but started thinking I was making too many changes.

jernoble commented on Source/WebCore/WebCore.xcodeproj/project.pbxproj:
```diff
@@ -3,7 +3,7 @@
 	classes = {
 	};
-	objectVersion = 54;
+	objectVersion = 51;
```
This should probably be removed from the final patch.

darinadler replied:
I guess, but why. Xcode did it, and someone else’s Xcode will undo it.

jernoble replied:
That's true; but I think it means the project file won't be able to be opened on older versions of Xcode. I know I've been asked to roll out this change in the past as well.

darinadler replied:
That’s backwards. My Xcode is changing the version back to an older version. The thing you were asked to revert was a roll _forward_. Seems super-unlikely that an _older_ `objectVersion` would prevent the file from being opened on _older_ versions of Xcode.

jernoble replied:
You're right! My mistake; I was pattern matching with the +-'s reversed.

jernoble commented on Source/WebCore/inspector/InspectorCanvas.h:
```diff
@@ -28,7 +28,6 @@
 #include "InspectorCanvasCallTracer.h"
 #include <JavaScriptCore/AsyncStackTrace.h>
 #include <JavaScriptCore/InspectorProtocolObjects.h>
-#include <JavaScriptCore/JSCInlines.h>
```
:heart:

jernoble reviewed (approved):
r=me. Minor nit around the .pbxproj.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ee0f3f161c2feb5bb9157f3d77de1ba06a7e16fb. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/11083

darinadler committed:
Cut the size of JSGlobalObject.h down
https://bugs.webkit.org/show_bug.cgi?id=253376
rdar://problem/106225153

Reviewed by Jer Noble.

Moved things out of JSGlobalObject.h, and removed unneeded includes of JSGlobalObject.h.

* Source/JavaScriptCore/API/APICast.h: Updated includes.

* Source/JavaScriptCore/API/JSAPIGlobalObject.cpp:
(JSC::JSAPIGlobalObject::globalObjectMethodTable): Use function for this class's
globalObjectMethodTable so header doesn't need to include GlobalObjectMethodTable.h.
* Source/JavaScriptCore/API/JSAPIGlobalObject.h: Ditto.
* Source/JavaScriptCore/API/JSAPIGlobalObject.mm:
(JSC::JSAPIGlobalObject::globalObjectMethodTable): Ditto.

* Source/JavaScriptCore/API/JSCallbackObjectFunctions.h: Updated includes.

* Source/JavaScriptCore/API/JSClassRef.cpp:
(OpaqueJSClass::contextData): Use new JSGlobalObject::contextData function, helping hide
more of the details of the JSGlobalObject rare data from other includers of JSGlobalObject.h

* Source/JavaScriptCore/CMakeLists.txt: Added GlobalObjectMethodTable.h.
* Source/JavaScriptCore/JavaScriptCore.xcodeproj/project.pbxproj: Ditto.

* Source/JavaScriptCore/bytecode/BytecodeIntrinsicRegistry.cpp: Updated includes.
* Source/JavaScriptCore/bytecode/CodeBlock.h: Ditto.
* Source/JavaScriptCore/bytecode/ExecutionCounter.h: Ditto.
* Source/JavaScriptCore/bytecode/ObjectAllocationProfile.h: Ditto.
* Source/JavaScriptCore/bytecode/SpeculatedType.cpp: Ditto.
* Source/JavaScriptCore/bytecompiler/NodesCodegen.cpp: Ditto.
* Source/JavaScriptCore/debugger/DebuggerEvalEnabler.h: Ditto.
* Source/JavaScriptCore/debugger/ScriptProfilingScope.h: Ditto.
* Source/JavaScriptCore/dfg/DFGOperations.cpp: Ditto.
* Source/JavaScriptCore/dfg/DFGStrengthReductionPhase.cpp: Ditto.
* Source/JavaScriptCore/ftl/FTLAbstractHeapRepository.cpp: Ditto.
* Source/JavaScriptCore/heap/Heap.cpp: Ditto.

* Source/JavaScriptCore/heap/IsoSubspace.h: Removed unused isAllocatedFromIsoSubspace.

* Source/JavaScriptCore/interpreter/CallFrameInlines.h: Updated includes.
* Source/JavaScriptCore/interpreter/Interpreter.cpp: Ditto.
* Source/JavaScriptCore/jsc.cpp: Ditto.
* Source/JavaScriptCore/runtime/AggregateError.h: Ditto.
* Source/JavaScriptCore/runtime/AggregateErrorConstructor.cpp: Ditto.
* Source/JavaScriptCore/runtime/AggregateErrorConstructor.h: Ditto.
* Source/JavaScriptCore/runtime/AggregateErrorPrototype.h: Ditto.
* Source/JavaScriptCore/runtime/ArrayBufferSharingMode.h: Ditto.
* Source/JavaScriptCore/runtime/ArrayPrototypeInlines.h: Ditto.
* Source/JavaScriptCore/runtime/BytecodeCacheError.h: Ditto.
* Source/JavaScriptCore/runtime/DeferredWorkTimer.cpp: Ditto.
* Source/JavaScriptCore/runtime/DirectEvalExecutable.cpp: Ditto.
* Source/JavaScriptCore/runtime/ExecutableBase.h: Ditto.
* Source/JavaScriptCore/runtime/FunctionConstructor.cpp: Ditto.

* Source/JavaScriptCore/runtime/GlobalObjectMethodTable.h: Added. Code here was moved
from JSGlobalObject.h.

* Source/JavaScriptCore/runtime/IndirectEvalExecutable.cpp: Updated includes.
* Source/JavaScriptCore/runtime/IntlObject.cpp: Ditto.

* Source/JavaScriptCore/runtime/JSArrayBufferView.h: Marked some functions inline so we
get warnings if we forget to include the JSArrayBufferViewInlines.h header, even in
debug builds.

* Source/JavaScriptCore/runtime/JSBigInt.h: Updated includes.
* Source/JavaScriptCore/runtime/JSFinalizationRegistry.cpp: Ditto.

* Source/JavaScriptCore/runtime/JSGenericTypedArrayView.h: Moved inline function bodies
out of here into JSGenericTypedArrayViewInlines.h.
* Source/JavaScriptCore/runtime/JSGenericTypedArrayViewInlines.h: Ditto.

* Source/JavaScriptCore/runtime/JSGlobalObject.cpp:
(JSC::JSGlobalObject::baseGlobalObjectMethodTable): Use function for this so header
doesn't need to include GlobalObjectMethodTable.h.
(JSC::JSGlobalObject::JSGlobalObject): Use baseGlobalObjectMethodTable.
(JSC::JSGlobalObject::typedArrayConstructorSpeciesAbsenceWatchpoint): Moved from header.
It is only used in this file.
(JSC::JSGlobalObject::typedArrayPrototypeSymbolIteratorAbsenceWatchpoint): Ditto.
(JSC::JSGlobalObject::typedArrayPrototypeConstructorWatchpoint): Ditto.

* Source/JavaScriptCore/runtime/JSGlobalObject.h: Updated includes and forward declarations.
Moved GlobalObjectData out of the header. Renamed JSGlobalObject::JSGlobalObjectRareData
to JSGlobalObject::RareData and moved it to the inlines header. Moved
JSGlobalObject::GlobalPropertyInfo out of the header.

* Source/JavaScriptCore/runtime/JSGlobalObjectFunctions.cpp: Updated includes.

* Source/JavaScriptCore/runtime/JSGlobalObjectInlines.h: Moved things here from JSGlobalObject.h.

* Source/JavaScriptCore/runtime/JSModuleLoader.cpp: Updated includes.
* Source/JavaScriptCore/runtime/JSONObject.cpp: Ditto.
* Source/JavaScriptCore/runtime/JSObjectInlines.h: Ditto.
* Source/JavaScriptCore/runtime/JSPromise.cpp: Ditto.
* Source/JavaScriptCore/runtime/JSScriptFetchParameters.h: Ditto.
* Source/JavaScriptCore/runtime/JSScriptFetcher.h: Ditto.
* Source/JavaScriptCore/runtime/JSSourceCode.h: Ditto.
* Source/JavaScriptCore/runtime/JSWithScope.h: Ditto.
* Source/JavaScriptCore/runtime/LazyClassStructure.h: Ditto.
* Source/JavaScriptCore/runtime/LazyProperty.h: Ditto.
* Source/JavaScriptCore/runtime/ModuleProgramExecutable.h: Ditto.
* Source/JavaScriptCore/runtime/PrototypeKey.h: Ditto.
* Source/JavaScriptCore/runtime/ProxyObject.h: Ditto.

* Source/JavaScriptCore/runtime/RegExpCachedResult.h: Moved RegExpCachedResult::record
to RegExpGlobalDataInlines.h.
* Source/JavaScriptCore/runtime/RegExpGlobalDataInlines.h: Ditto.

* Source/JavaScriptCore/runtime/ShadowRealmObject.cpp: Updated includes.
* Source/JavaScriptCore/runtime/StructureCache.h: Ditto.
* Source/JavaScriptCore/runtime/StructureInlines.h: Ditto.
* Source/JavaScriptCore/tools/JSDollarVM.cpp: Ditto.
* Source/JavaScriptCore/wasm/js/JSWebAssembly.cpp: Ditto.
* Source/JavaScriptCore/wasm/js/JSWebAssemblyHelpers.h: Ditto.

* Source/WebCore/Headers.cmake: Removed WebGLAnyInlines.h.

* Source/WebCore/Modules/encryptedmedia/legacy/LegacyCDMSessionClearKey.cpp:
Updated includes.

* Source/WebCore/WebCore.xcodeproj/project.pbxproj: Removed WebGLAnyInlines.h.

* Source/WebCore/bindings/js/JSCallbackData.h: Updated includes.
* Source/WebCore/bindings/js/JSDOMConvertBufferSource.h: Ditto.
* Source/WebCore/bindings/js/JSDOMConvertDate.h: Ditto.
* Source/WebCore/bindings/js/JSDOMConvertUnion.h: Ditto.
* Source/WebCore/bindings/js/JSDOMGlobalObject.cpp: Ditto.
* Source/WebCore/bindings/js/JSDOMGlobalObject.h: Ditto.
* Source/WebCore/bindings/js/JSDOMGuardedObject.h: Ditto.

* Source/WebCore/bindings/js/JSDOMWindowBase.cpp:
(WebCore::JSDOMWindowBase::globalObjectMethodTable): Use function for this class's
globalObjectMethodTable so header doesn't need to include GlobalObjectMethodTable.h.
* Source/WebCore/bindings/js/JSDOMWindowBase.h: Ditto.

* Source/WebCore/bindings/js/JSDOMWrapperCache.h: Updated includes.

* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.cpp:
(WebCore::JSRemoteDOMWindowBase::globalObjectMethodTable): Use function for this class's
globalObjectMethodTable so header doesn't need to include GlobalObjectMethodTable.h.
* Source/WebCore/bindings/js/JSRemoteDOMWindowBase.h: Ditto.
* Source/WebCore/bindings/js/JSShadowRealmGlobalScopeBase.cpp:
(WebCore::JSShadowRealmGlobalScopeBase::globalObjectMethodTable): Ditto.
* Source/WebCore/bindings/js/JSShadowRealmGlobalScopeBase.h: Ditto.
* Source/WebCore/bindings/js/JSWorkerGlobalScopeBase.cpp:
(WebCore::JSWorkerGlobalScopeBase::globalObjectMethodTable): Ditto.
* Source/WebCore/bindings/js/JSWorkerGlobalScopeBase.h: Ditto.
* Source/WebCore/bindings/js/JSWorkletGlobalScopeBase.cpp:
(WebCore::JSWorkletGlobalScopeBase::globalObjectMethodTable): Ditto.
* Source/WebCore/bindings/js/JSWorkletGlobalScopeBase.h: Ditto.

* Source/WebCore/bindings/js/ScriptWrappable.h: Add inline keyword for declaration of
some functions that are inlines in the ScriptWrappableInlines.h header.

* Source/WebCore/bindings/js/ScriptWrappableInlines.h: Updated includes.
* Source/WebCore/bindings/js/SerializedScriptValue.cpp: Ditto.
* Source/WebCore/bindings/js/WebCoreTypedArrayController.h: Ditto.
* Source/WebCore/bridge/runtime_method.h: Ditto.
* Source/WebCore/html/FormAssociatedCustomElement.h: Ditto.
* Source/WebCore/html/canvas/EXTDisjointTimerQuery.cpp: Ditto.
* Source/WebCore/html/canvas/WebGL2RenderingContext.cpp: Ditto.

* Source/WebCore/html/canvas/WebGLAnyInlines.h: Removed.

* Source/WebCore/html/canvas/WebGLRenderingContext.cpp: Updated includes.
* Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp: Ditto.
* Source/WebCore/html/canvas/WebGLRenderingContextBase.h: Ditto.
* Source/WebCore/html/track/DataCue.h: Ditto.
* Source/WebCore/inspector/InspectorCanvas.h: Ditto.
* Source/WebCore/page/WindowOrWorkerGlobalScope.cpp: Ditto.
* Source/WebCore/platform/SerializedPlatformDataCue.h: Ditto.
* Source/WebKit/Platform/IPC/JSIPCBinding.h: Ditto.

Canonical link: https://commits.webkit.org/261249@main

webkit-commit-queue commented:
Committed 261249@main (568657736a63): <https://commits.webkit.org/261249@main>

Reviewed commits have been landed. Closing PR #11083 and removing active labels.
